packages/opentelemetry-core/src/trace/GlobalTracerDelegate.ts

```diff
@@ -17,18 +17,35 @@
 import * as types from '@opentelemetry/types';
 import { NoopTracer } from './NoopTracer';
 
-const noopTracer = new NoopTracer();
-let _globalTracer: types.Tracer | null = null;
-
+// Acts a bridge to the global tracer that can be safely called before the
+// global tracer is initialized. The purpose of the delegation is to avoid the
+// sometimes nearly intractible initialization order problems that can arise in
+// applications with a complex set of dependencies. Also allows for the tracer
+// to be changed/disabled during runtime without needing to change reference
+// to the global tracer
 export class GlobalTracerDelegate implements types.Tracer {
```

Comment:
What would you think about calling this class just `TracerDelegate` since the global aspect would be handled by different packages exporting their own tracer singletons?